storage/local/storage.go

```diff
@@ -360,11 +371,28 @@ func (s *memorySeriesStorage) GetMetricForFingerprint(fp clientmodel.Fingerprint
 
 // AppendSamples implements Storage.
 func (s *memorySeriesStorage) AppendSamples(samples clientmodel.Samples) {
+	if len(samples) == 0 {
```

Comment:
Ahhh... the empty slices weren't coming from the scrapes at all. They were coming from rules which had empty results. I think we should just not send empty result sets into the ingestion channel from the rule evaluator.